src/main/java/service/StudentService.java

```diff
@@ -0,0 +1,36 @@
+package service;
+
+import entity.Student;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class StudentService {
+
+
+    public Double averageRatingForSubject(List<Student> students, String subjectName){
+        return students.stream().map(Student::getRating).mapToDouble(d -> d.get(subjectName))
```

Comment:
* every new method execution in stream chain write in next line
* while lambda initialization write more readable name, not 'd' etc.